src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -183,6 +183,25 @@ static Value builtinStringSubstr(ExecutionState& state, Value thisValue, size_t
 
 static Value builtinStringMatch(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
 {
+#ifdef ESCARGOT_ENABLE_ES2015
+    RESOLVE_THIS_BINDING_TO_OBJECT(obj, Object, match);
+    Value argument = argv[0];
+    RegExpObject* regexp;
+
+    if (argument.isUndefinedOrNull() == false) {
+        Value match = obj->getMethod(state, argument, ObjectPropertyName(state, state.context()->vmInstance()->globalSymbols().match));
```

Comment:
`Let matcher be GetMethod(regexp, @@match).` You getting the method of `obj` instead of the `regexp` (1st argument).